projects/backend/src/main/java/backend/security/UserDetailsServiceImpl.java

```diff
@@ -0,0 +1,30 @@
+package backend.security;
+
+import static java.util.Collections.emptyList;
+
+import backend.repositories.UsersRepository;
+import org.springframework.security.core.userdetails.User;
+import org.springframework.security.core.userdetails.UserDetails;
+import org.springframework.security.core.userdetails.UserDetailsService;
+import org.springframework.security.core.userdetails.UsernameNotFoundException;
+import org.springframework.stereotype.Service;
+
+
+@Service
+public class UserDetailsServiceImpl implements UserDetailsService {
+  private UsersRepository applicationUserRepository;
+
+  public UserDetailsServiceImpl(UsersRepository applicationUserRepository) {
+    this.applicationUserRepository = applicationUserRepository;
+  }
+
+  @Override
+  public UserDetails loadUserByUsername(String username) throws UsernameNotFoundException {
+    backend.domain.User applicationUser = applicationUserRepository
+        .findByIndexNumber(username).get();
```

Comment:
Instead of ``` backend.domain.User``` (vary long type, cause I guess there is a different User type) you can use ```var``` for brewity